### PR TITLE
Use getX() to get values from subclasses instead of cast.

### PR DESCRIPTION
**Pull request by songy23**
Fixes https://github.com/bogdandrutu/openconsensus/issues/70.

### CODE REVIEW

File: api/src/main/java/openconsensus/metrics/data/Value.java [hunk 1]
```diff
@@ -86,6 +86,54 @@ public static Value summaryValue(Summary value) {
...
 
+  /**
+   * Returns the double value of this {@link Value}.
+   *
+   * <p>This method should only be called with {@link Type#DOUBLE} {@link Value}.
+   *
+   * @return the double {@code Value}.
+   * @since 0.1.0
+   */
+  public double getDouble() {
```
Comment (bogdandrutu) [inline api/src/main/java/openconsensus/metrics/data/Value.java]: To be consistent with AttributeValue I would suggest getDoubleValue()

Comment (songy23) [reply api/src/main/java/openconsensus/metrics/data/Value.java]: SG, updated.

File: api/src/main/java/openconsensus/metrics/data/Distribution.java [hunk 2]
```diff
@@ -166,6 +166,18 @@ public static BucketOptions explicitOptions(List<Double> bucketBoundaries) {
...
+    /**
+     * Returns the bucket boundaries of this distribution.
+     *
+     * <p>Currently only {@link ExplicitOptions} implements this method.
+     *
+     * @return the bucket boundaries of this distribution.
+     * @since 0.1.0
+     */
+    public List<Double> getBucketBoundaries() {
+      throw new UnsupportedOperationException("Not implemented.");
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/metrics/data/Distribution.java]: nit: message may say - "Only ExplicitOptions implements this method"

File: api/src/main/java/openconsensus/metrics/data/Value.java [hunk 3]
```diff
@@ -86,6 +86,54 @@ public static Value summaryValue(Summary value) {
...
+  /**
+   * Returns the double value of this {@link Value}.
+   *
+   * <p>This method should only be called with {@link Type#DOUBLE} {@link Value}.
+   *
+   * @return the double {@code Value}.
+   * @since 0.1.0
+   */
+  public double getDoubleValue() {
+    throw new UnsupportedOperationException("Not implemented.");
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/metrics/data/Value.java]: "not implemented" makes wrong impression we want to implement it. Perhaps change the message to "This type can only return Foo data"

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/metrics/data/Value.java]: String.format("This type can only return %s data", getType().name())??

File: api/src/main/java/openconsensus/metrics/data/Distribution.java (hunk 2 above)
Comment (songy23) [reply api/src/main/java/openconsensus/metrics/data/Distribution.java]: Updated.

File: api/src/main/java/openconsensus/metrics/data/Value.java (hunk 3 above)
Comment (songy23) [reply api/src/main/java/openconsensus/metrics/data/Value.java]: SGTM, updated.

**Pull request merged**